File: web/src/types/adminImport.ts
export type ImportType = 'strategy_daily_returns' | 'strategy_signals'
export type ImportMode = 'upsert' | 'insert_only'
export type ReturnCalcMethod = 'compound' | 'simple'

export interface StrategyUpsertBody {
  strategy_code: string
  name: string
  strategy_type: string
  summary?: string | null
  description?: string | null
  detail_html?: string | null
  category_id?: string | null
  asset_class: string
  market: string
  risk_level: 'low' | 'medium' | 'high'
  run_status: string
  pub_status: string
  author_id: string
  cover_image?: string | null
  subscription_monthly: string
  subscription_yearly: string
  backtest_start?: string | null
  backtest_end?: string | null
}

export interface StrategyUpsertData {
  strategy_id: string
  strategy_code: string
  name: string
  pub_status: string
  run_status: string
}

export interface ImportPreviewBody {
  import_type: ImportType
  csv_text: string
  file_name: string
  strategy_code?: string | null
  mode: ImportMode
  return_calc_method: ReturnCalcMethod
  initial_nav: number
  trading_days_per_year: number
  risk_free_rate: number
}

export interface ImportSummary {
  total_rows: number
  valid_rows: number
  error_rows: number
  duplicate_rows: number
  will_insert: number
  will_update: number
  return_calc_method?: ReturnCalcMethod
  initial_nav?: number
  trading_days_per_year?: number
  risk_free_rate?: number
  derived_equity_rows?: number
  derived_monthly_rows?: number
  derived_snapshot_rows?: number
  affected_rows?: number
}

export interface ImportErrorRow {
  row_number: number
  column: string | null
  error_code: string
  message: string
  raw_row: Record<string, unknown>
}

export interface ImportPreviewData {
  job_id: string
  status: 'validated' | 'blocked'
  summary: ImportSummary
  preview_rows: Record<string, unknown>[]
  errors: ImportErrorRow[]
}

export interface ImportCommitData {
  job_id: string
  status: 'committed'
  affected_rows: number
}

export interface ImportJob {
  job_id: string
  import_type: ImportType
  file_name: string
  mode: ImportMode
  status: 'validated' | 'blocked' | 'committed'
  summary: ImportSummary
  created_by: string
  committed_by: string | null
  created_at: string
  committed_at: string
}

export interface ImportHistoryData {
  list: ImportJob[]
}

File: packages/gr-backtest/src/getrich/apps/web/schemas/admin_import.py
"""后台导入模块请求体模型。"""

from __future__ import annotations

from datetime import date
from typing import Literal

from pydantic import BaseModel, Field


ImportType = Literal["strategy_daily_returns", "strategy_signals"]
ImportMode = Literal["upsert", "insert_only"]
ReturnCalcMethod = Literal["compound", "simple"]


class StrategyUpsertIn(BaseModel):
    """策略基础信息创建或更新请求体。"""

    strategy_code: str = Field(min_length=3, max_length=64)
    name: str = Field(min_length=1, max_length=128)
    strategy_type: str = Field(default="manual", min_length=1, max_length=32)
    summary: str | None = Field(default=None, max_length=500)
    description: str | None = None
    detail_html: str | None = None
    category_id: str | None = Field(default=None, max_length=64)
    asset_class: str = Field(min_length=1, max_length=32)
    market: str = Field(default="cn", min_length=1, max_length=32)
    risk_level: Literal["low", "medium", "high"]
    run_status: str = Field(default="paper", max_length=32)
    pub_status: str = Field(default="draft", max_length=32)
    author_id: str
    cover_image: str | None = Field(default=None, max_length=512)
    subscription_monthly: str = Field(default="0")
    subscription_yearly: str = Field(default="0")
    backtest_start: date | None = None
    backtest_end: date | None = None


class ImportPreviewIn(BaseModel):
    """CSV 导入预检请求体。"""

    import_type: ImportType
    csv_text: str = Field(min_length=1)
    file_name: str = Field(min_length=1, max_length=256)
    strategy_code: str | None = Field(default=None, max_length=64)
    mode: ImportMode = "upsert"
    return_calc_method: ReturnCalcMethod = "compound"
    initial_nav: float = Field(default=1.0, gt=0)
    trading_days_per_year: int = Field(default=252, ge=1, le=366)
    risk_free_rate: float = 0.0
